### PR TITLE
Add example of the setPattern method usage

### PR DESCRIPTION
**Pull request by anasmi**
Fixes https://github.com/vaadin/vaadin-text-field-flow/issues/3 https://github.com/vaadin/90min-dx-tasks/issues/104

### CODE REVIEW

File: vaadin-text-field-flow-demo/src/main/java/com/vaadin/flow/component/textfield/demo/TextFieldView.java [hunk 1]
```diff
@@ -164,6 +165,31 @@ private void textFieldFocusShortcut() {
...
+        Paragraph status=new Paragraph();
+        TextField username = new TextField("Username");
+        username.setPattern("^[a-zA-Z0-9._-]{3,}");
+        username.addValueChangeListener(e->{
+            if(username.isInvalid())
+                status.setText("Username should be, at least, 3 character long " +
+                        "and contain only letters, digits, dashes or dots.");
+            else
+                status.setText("Your username seems valid!");
+        });
```
Comment (tomivirkki) [inline vaadin-text-field-flow-demo/src/main/java/com/vaadin/flow/component/textfield/demo/TextFieldView.java]: Lots of formatting issues in this example, please use an autoformatter

Comment (anasmi) [reply vaadin-text-field-flow-demo/src/main/java/com/vaadin/flow/component/textfield/demo/TextFieldView.java]: is there some maven goal for this? Or do you have rules somewhere?

Comment (tomivirkki) [reply vaadin-text-field-flow-demo/src/main/java/com/vaadin/flow/component/textfield/demo/TextFieldView.java]: Here are the conventions https://github.com/vaadin/flow/tree/master/eclipse

Comment (anasmi) [reply vaadin-text-field-flow-demo/src/main/java/com/vaadin/flow/component/textfield/demo/TextFieldView.java]: Done

**Pull request merged**